Realm/Realm/DatabaseTypes/RealmDictionary.cs

```diff
@@ -98,6 +98,19 @@ public ICollection<TValue> Values
             }
         }
 
+        internal RealmResults<TValue> ToResults()
+        {
+            var resultsHandle = _dictionaryHandle.GetValues();
+            return new RealmResults<TValue>(Realm, resultsHandle, Metadata);
+        }
```

Comment:
This is the same as `Values` - we could just use that and cast to `RealmResults<T>`.

Reply:
I did have a comment mentioning this, is this weird casting to Collection and then casting to Results again a good idea?

Reply:
Yeah, I think it should be fine - as long as we have tests for it, any refactoring that breaks assumptions will immediately show up as a failed test.